user/caribou/module/src/H2MEventConverter.cc

```diff
@@ -143,3 +185,49 @@ bool H2MEvent2StdEventConverter::Converting(
   // Indicate that data was successfully converted
   return true;
 }
+
+void H2MEvent2StdEventConverter::loadCalibration(std::string path, char delim, std::vector<std::vector<float>>& dat) const {
+    std::ifstream f;
+    f.open(path);
+    dat.clear();
+
+    // check if file is open
+    if(!f.is_open()) {
+        throw DataInvalid("Cannot open calibration file:\n\t" + path);
+        return;
+    }
+
+    // read file line by line
+    int i = 0;
+    std::string line;
+    while(!f.eof()) {
+        std::getline(f, line);
+
+        // check if line is empty or a comment
+        // if not write to output vector
+
+        if(line.size() > 0 && isdigit(line.at(0))) {
+            std::stringstream ss(line);
+            std::string word;
+            std::vector<float> row;
+            for (size_t i = 0; i < 6; ++i) {
+                std::getline(ss, word, delim);
+                if (i >= 2) { // Keep only calibration data
+                    row.push_back(std::stof(word));
```

Comment:
Tiny thing, and just because I wanted to mention this - now you discard the addresses but never check them. Imagine (for some arcane reason):

```
0 1 9 9 9 9
0 3 9 9 9 8
0 2 9 9 9 7
```

would mean that you accidentally swap the values of pixel (0, 2) and (0, 3) because you assume they are in the file in the correct order.

...but let's not get overboard with nitpicky here. :wink: